management/server/sqlite_store.go

```diff
@@ -188,18 +211,38 @@ func (s *SqliteStore) SaveAccount(account *Account) error {
 
 		result = tx.
 			Session(&gorm.Session{FullSaveAssociations: true}).
-			Clauses(clause.OnConflict{UpdateAll: true}).Create(account)
+			Clauses(clause.OnConflict{UpdateAll: true}).
+			Omit("PeersG", "GroupsG", "UsersG", "SetupKeysG", "RoutesG").
+			Create(account)
 		if result.Error != nil {
 			return result.Error
 		}
-		return nil
+
+		const batchSize = 500
+		err := batchInsert(account.PeersG, batchSize, tx)
+		if err != nil {
+			return err
```

Comment:
Can we add context to all these, to help identify potential issues? We can probably deduce the location from the returned error already, but I'm not sure.
 
```suggestion
			return fmt.Errorf("insert PeersG: %w", err)
```